### ISSUE TITLE
entry functions don't work in loop

### ISSUE THREAD

**Issue by januz** [ropenscilabs/vitae#43]
I would like to generate a CV programmatically, i.e., by reading out data from a spreadsheet, generating a data frame / tibble that contains (sub)heading titles and the corresponding data, and then looping over this data frame within an RMarkdown code chunk. My goal is to provide this workflow to someone without programming knowledge so that (s)he can just fill out the spreadsheet, add/remove sheets to add sections, etc. and have the CV being generated without ever having to write `R` or `RMarkdown` code.

Unfortunately, using the `detailed_entries()`/`simple_entries()` functions doesn't seem to work within a loop. It would be great if you point me in the right direction.

Here a simplified version (that doesn't parse a spreadsheet, but just feeds a dataframe with section names and data to RMarkdown):

file `compile.R`

```r
library(tidyverse)
library(vitae)
library(pander)
library(rmarkdown)

edu <- rorcid::orcid_educations("0000-0001-6729-7695")
df_education <- edu$`0000-0001-6729-7695`$`education-summary` %>%
  transmute(
    what = `role-title`,
    when = glue::glue("{`start-date.year.value`} - {`end-date.year.value`}"),
    with = organization.name,
    where = organization.address.city,
    why = NA
  ) %>% 
  as_tibble()

df_packages <- pkgsearch::ps("O'Hara-Wild",size = 100) %>%
  filter(purrr::map_lgl(package_data, ~ grepl("Mitchell O'Hara-Wild", .x$Author, fixed = TRUE))) %>%
  arrange(desc(downloads_last_month)) %>%
  transmute(
    what = title,
    when = lubridate::year(date),
    with = description
  )

sections <- tribble(
  ~name, ~data,
  "Education", df_education,
  "Packages", df_packages
)

render("CV.Rmd")
```

file `CV.Rmd`

````
---
name: Mitchell O'Hara-Wild
date: "`r format(Sys.time(), '%B, %Y')`"
qualifications: BCom(Hons), BSc
position: Data Scientist
address: 58 Madeleine Rd, Clayton
phone: +61 408 259 421
email: mail@mitchelloharawild.com
website: mitchelloharawild.com
github: mitchelloharawild
linkedin: mitchelloharawild
twitter: mitchoharawild
headcolor: 414141
output: vitae::awesomecv
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

```{r, results='asis'}
for (i in 1:nrow(sections)){
  row <- slice(sections, i)
  
  pandoc.header(row$name, level = 1, "atx")
  
  data <- row$data[[1]]
  if (ncol(data) == 3){
    data %>% 
      brief_entries(what, when, with)
  } else {
    data %>% 
      detailed_entries(what, when, with, where, why)
  }
  
  cat("\n\n")
}
```
````

**Comment by mitchelloharawild** [comment]
It works fine. The issue is that the entries functions return tibbles, and not formatted output suitable for results='asis'. When in a loop, objects don't print automatically, so the CV entries are not being printed in the first place. To print the entries in a suitable format for 'asis' output, you would need to use `knitr::knit_print()`. You would also need to ensure that they are actually displayed with `cat`.

`CV.Rmd`
````r
---
name: Mitchell O'Hara-Wild
date: "`r format(Sys.time(), '%B, %Y')`"
qualifications: BCom(Hons), BSc
position: Data Scientist
address: 58 Madeleine Rd, Clayton
phone: +61 408 259 421
email: mail@mitchelloharawild.com
website: mitchelloharawild.com
github: mitchelloharawild
linkedin: mitchelloharawild
twitter: mitchoharawild
headcolor: 414141
output: vitae::awesomecv
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```


```{r, results='asis'}
for (i in 1:nrow(sections)){
  row <- slice(sections, i)
  
  pandoc.header(row$name, level = 1, "atx")
  
  data <- row$data[[1]]
  if (ncol(data) == 3){
    data %>%
      brief_entries(what, when, with) %>% 
      knitr::knit_print() %>% 
      cat
  } else {
    data %>%
      detailed_entries(what, when, with, where, why) %>% 
      knitr::knit_print() %>% 
      cat
  }
}
```
````
`compile.R` has also been changed to remove `\n` from the package descriptions.
```r
library(tidyverse)
library(vitae)
library(pander)
library(rmarkdown)

edu <- rorcid::orcid_educations("0000-0001-6729-7695")
df_education <- edu$`0000-0001-6729-7695`$`education-summary` %>%
  transmute(
    what = `role-title`,
    when = glue::glue("{`start-date.year.value`} - {`end-date.year.value`}"),
    with = organization.name,
    where = organization.address.city,
    why = NA
  ) %>%
  as_tibble()

df_packages <- pkgsearch::ps("O'Hara-Wild",size = 100) %>%
  filter(purrr::map_lgl(package_data, ~ grepl("Mitchell O'Hara-Wild", .x$Author, fixed = TRUE))) %>%
  arrange(desc(downloads_last_month)) %>%
  transmute(
    what = title,
    when = lubridate::year(date),
    with = gsub("\n", "", description)
  ) %>%
  as_tibble # drop the pkg_search class

sections <- tribble(
  ~name, ~data,
  "Education", df_education,
  "Packages", df_packages
)

render("loop/CV.Rmd")
```

**Comment by januz** [comment]
Thank you so much for your help! This solves my problem.

I now recognized another, more cosmetic, issue, though: The second and following entries within a category are slightly more indented than the first one. Is that a general problem? (I just checked your CV on Github and it also seems to have the issue; https://github.com/mitchelloharawild/CV/blob/master/CV.pdf).

Thanks!

![image](https://user-images.githubusercontent.com/1440498/52831829-fbd26300-308a-11e9-9bed-607de923b9be.png)
